### PR TITLE
add metadata env vars

### PR DESCRIPTION
**Pull request by nsinkov**
## Changes proposed in this PR

- add metadata env vars

## Why are we making these changes?

We need access to more metadata like cook URL to get info about past instances, compute cluster name, pool name

### CODE REVIEW

File: scheduler/src/cook/kubernetes/api.clj [hunk 1]
```diff
@@ -800,7 +800,10 @@
         params-env (build-params-env parameters)
         progress-env (task/build-executor-environment job)
         checkpoint-env (checkpoint->env checkpoint)
-        main-env-base (merge environment params-env progress-env sandbox-env checkpoint-env)
+        metadata-env (cond-> {"COOK_COMPUTE_CLUSTER_NAME" compute-cluster-name
+                              "COOK_SCHEDULER_REST_URL" (config/scheduler-rest-url)}
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj:804]: Why not just `COOK_SCHEDULER_URL`?

File: scheduler/src/cook/kubernetes/api.clj [hunk 2]
```diff
@@ -800,7 +800,10 @@
...
+                       pool-name (assoc "COOK_POOL" pool-name))
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj:805]: Will this be `nil` if the user doesn't specify a pool (and lands in the default pool)? I assume we want to make it resolve to the name of the default pool?

File: scheduler/src/cook/kubernetes/api.clj [hunk 3]
```diff
@@ -800,7 +800,10 @@
...
+        main-env-base (merge environment params-env progress-env sandbox-env checkpoint-env metadata-env)
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj:806]: Would it make sense to write an integration test for the URL and POOL environment variables?

Comment (nsinkov) [reply scheduler/src/cook/kubernetes/api.clj:807]: added

File: scheduler/src/cook/kubernetes/api.clj [hunk 4]
```diff
@@ -800,7 +801,10 @@
         params-env (build-params-env parameters)
         progress-env (task/build-executor-environment job)
         checkpoint-env (checkpoint->env checkpoint)
-        main-env-base (merge environment params-env progress-env sandbox-env checkpoint-env)
+        metadata-env {"COOK_COMPUTE_CLUSTER_NAME" compute-cluster-name
+                      "COOK_POOL" (pool/pool-name-or-default pool-name)
+                      "COOK_SCHEDULER_REST_URL" (config/scheduler-rest-url)}
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:806]: COOK_SCHEDULER_REST_URL is already used by the sidecar. can't rename to COOK_SCHEDULER_URL right now

File: integration/tests/cook/test_basic.py [hunk 5]
```diff
@@ -3226,6 +3226,25 @@ def test_kubernetes_checkpointing(self):
...
+    def test_kubernetes_metadata_env_vars(self):
+        docker_image = util.docker_image()
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image}}
+        default_pool = util.default_submit_pool() or util.default_pool(self.cook_url)
+        # info = util.scheduler_info(self.cook_url)
+        try:
+            # need to figure out how to match up COOK_SCHEDULER_REST_URL
+            # leader url and COOK_SCHEDULER_REST_URL can have different schemes right now
+            #command = 'bash -c \'if [[ "${COOK_POOL:-zzz}" == "' + default_pool + '" ]] && [[ "${COOK_SCHEDULER_REST_URL:-zzz}" == "' + info['leader-url'] + '" ]]; then exit 0; else exit 1; fi\''
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:3239]: Remove commented out lines?

Comment (nsinkov) [reply integration/tests/cook/test_basic.py:3239]: I have them as a TODO

Comment (dposada) [reply integration/tests/cook/test_basic.py:3239]: Should we compare everything except the scheme?

Comment (nsinkov) [reply integration/tests/cook/test_basic.py:3239]: it would be scheme and port.

Comment (dposada) [reply integration/tests/cook/test_basic.py:3239]: OK. I have a slight preference for removing the commented out code lines, but if you feel strongly that it's better with them there, that's fine.

File: integration/tests/cook/test_basic.py [hunk 6]
```diff
@@ -3226,6 +3226,25 @@ def test_kubernetes_checkpointing(self):
...
+            command = 'bash -c \'if [[ "${COOK_POOL:-zzz}" == "' + default_pool + '" ]] && [[ "${COOK_SCHEDULER_REST_URL:-zzz}" != "zzz" ]]; then exit 0; else exit 1; fi\''
+            job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+            self.assertEqual(201, resp.status_code, resp.text)
+            util.wait_for_instance(self.cook_url, job_uuid, status='success')
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:3243]: If we're going to wait for success, I think we need to add some retries to be resilient to random failures

Comment (nsinkov) [reply integration/tests/cook/test_basic.py:3243]: so going forward for new tests we want retries on any job submission? is there an example of this on a different test? can we implement retries on a test level instead, like in waiter?

Comment (dposada) [reply integration/tests/cook/test_basic.py:3243]: You only want the retries if you're expecting the test job to succeed (many tests don't have that expectation). Yeah, there are lots of examples, search for `max_retries=5` in the tests.

Comment (nsinkov) [reply integration/tests/cook/test_basic.py:3243]: oh, I wasn't thinking about cook retries. updated

**Review by dposada — APPROVED** [review]

**Comment by dposada** [comment]
Will merge when green

**Pull request merged**